### DISCUSSION TITLE
AttributeError: 'API' object has no attribute 'me'

### DISCUSSION THREAD

**Discussion by keyglow** [tweepy/tweepy#1696] [Questions] [answered]
Hello Guys, i wanna auto retweet stuff, but it show me everytime a Error.

`bot_id = int(api.me().id_str)
`

**AttributeError: 'API' object has no attribute 'me'**

```
import tweepy


consumer_key = 'ss' 
consumer_secret = 'ssssss'
access_token = 'ssss-sssss' 
access_token_secret = 'sssss'

auth = tweepy.OAuthHandler(consumer_key, consumer_secret)
auth.set_access_token(access_token, access_token_secret)
api = tweepy.API(auth)



bot_id = int(api.me().id_str)

class MyStreamListener(tweepy.StreamListener):
    def on_status(self, tweet):
        print(f"{tweet.author.screen_name} - {tweet.text}")


stream_listener = MyStreamListener()
stream = tweepy.Stream(auth=api.auth,listener=stream_listener)
stream.filter(track=["NBA"], language=["en"])
```

**Comment by keyglow** [comment]
**```
If you’re attempting to use API.me with Tweepy v4, you’ll get an AttributeError about the API object not having a me attribute.

This is because Tweepy v4.0.0 removed API.me.

Instead, you can use API.verify_credentials().
```**

So i did now ...

`bot_id = int(API.verify_credentials().id_str)`

doesnt work :/
   bot_id = int(API.verify_credentials().id_str)
NameError: name 'API' is not defined. Did you mean: 'api'?

**Reply by Harmon758** [reply]
[`tweepy.API`](https://tweepy.readthedocs.io/en/v4.2.0/api.html#tweepy.API) is the class.
You need to use the instance of it like you're doing in your original code or define `API`, as the error says.

**Comment by Harmon758** [comment, marked as answer]
You're using syntax from Tweepy v3 with Tweepy v4.
[Tweepy v4.0.0](https://github.com/tweepy/tweepy/releases/tag/v4.0.0) [removed `API.me`](https://tweepy.readthedocs.io/en/v4.2.0/faq.html#where-did-api-me-go) and [merged `StreamListener` into `Stream`](https://tweepy.readthedocs.io/en/v4.2.0/faq.html#where-did-streamlistener-go).
